### PR TITLE
Added MainView, TripsView, NewTripView, TripDetailView Screens

### PR DESCRIPTION
**Pull request by pavan3008**
# Proposed Changes

Added Landing Screen, TripsView, NewTripView, TripDetailView Screens using SwiftUI 

Screenshot 2023-02-15 at 2 34 35 PM Screenshot 2023-02-15 at 2 34 55 PM
Screenshot 2023-02-15 at 2 35 05 PM Screenshot 2023-02-15 at 2 35 23 PM

# Jira Stories

- [LAB-365](https://oregonstate-innovationlab.atlassian.net/browse/LAB-365)

# Checklist

- [x] Self-documenting code
- [x] This has been tested
- [x] Code is linted

# Instructions to review

Please review the code and recommend necessary changes

### CODE REVIEW

**Comment by pavan3008** [comment]
# Proposed Changes

Added Landing Screen, TripsView, NewTripView, TripDetailView Screens using SwiftUI 


Screenshot 2023-02-15 at 2 34 35 PM Screenshot 2023-02-15 at 2 34 55 PM
Screenshot 2023-02-15 at 2 35 05 PM Screenshot 2023-02-15 at 2 35 23 PM


# Jira Stories

- [LAB-365](https://oregonstate-innovationlab.atlassian.net/browse/LAB-365)

# Checklist

- [ ] This has been tested
- [ ] Self-documenting code
- [ ] Code is linted

# Instructions to review

Please review the code and recommend necessary changes

**Comment by NSR9** [comment]
Excellent everything looks good! The '+' button indentation has to be corrected. After correcting that go ahead and merge the PR.

**Comment by wbraynen** [comment]
@pavan3008 `features/feature1` is not a good name for a branch.  Names of things should be self documenting / self describing.  Same is true of your PR title: "Features/feature1".  Also please add a PR description describing the changes proposed in this PR.

Also, what jira ticket does this correspond to?

File: SmilesMiles/View/TripView.swift
Comment (wbraynen) [inline SmilesMiles/View/TripView.swift]: Do you have a linter installed?

You want consistency in formatting and don't want to waste PR time on reviewing how your code is formatted.  For example, this shoudl be `some View {` instead of `some View{`.  Elsewhere, code comments should read `// Blah` instead of `//Blah`.  (You use an inconsistent mix of formatting.)

File: SmilesMiles/View/TripView.swift
Comment (wbraynen) [inline SmilesMiles/View/TripView.swift]: Could you break it up to make it more self documenting?  That is:
```
let blah = trips.filter { searchText.isEmpty || $0.localizedCaseInsensitiveContains(searchText) }
ForEach(blah, id: \.self) {
```
instead of 
```
ForEach(trips.filter { searchText.isEmpty || $0.localizedCaseInsensitiveContains(searchText) }, id: \.self) {
```

Replace `blah` with a meaningful name.

File: SmilesMiles/View/TripView.swift
Comment (wbraynen) [inline SmilesMiles/View/TripView.swift]: A formatting nitpick: 

Why skip a line here and also why is line 42 intented relative to line 40?

File: SmilesMiles/View/TripView.swift
Comment (wbraynen) [inline SmilesMiles/View/TripView.swift]: Negative numbers for coordinates are usually trouble.  Are you sure it' really necessary?

Also, a formatting nitpick: why skip these lines?

File: SmilesMiles/View/TripView.swift
Comment (wbraynen) [inline SmilesMiles/View/TripView.swift]: `0.3700000047683716`—can you come up with a simpler number?  That seems unnecessarily precise?

Also, a formatting nitpick, but why `radius:24` instead of `radius: 24" when you skip a space after a colon elsewhere?

File: SmilesMiles/View/TripView.swift
Comment (wbraynen) [inline SmilesMiles/View/TripView.swift]: `0.20392157137393951`, `0.7803921699523926`, and `0.3490196168422699`—Can you find simpler numbers?

**Comment by pavan3008** [comment]
I've added the PR Template with an updated name and JIRA issue. For the branch name, should I rename it to **release/1.0.0**?

**Review by wbraynen — CHANGES_REQUESTED** [review]
Nice job.  Just a few nitpicks.

**Comment by pavan3008** [comment]
I've made all the recommended changes and the code is ready to merge.

File: SmilesMiles/View/TripView.swift
Comment (pavan3008) [reply SmilesMiles/View/TripView.swift]: I'm using SwiftLint for my project.

**Comment by pavan3008** [comment]
All the requested changes have been made along with the code lint. I don't see any other changes required and looks okay to be merged.

**Comment by wbraynen** [comment]
> All the requested changes have been made along with the code lint. I don't see any other changes required and looks okay to be merged.

@pavan3008 If so, then you want to re-request review.
image

It's also good to DM the person (me in this case) to say that it's back in their court and maybe slack 'em a link to the PR as a reminder.

To be clear: it's not to merge that you want.  What you want is for the person to review again and approve your PR.  You generally don't want to merge without approvals.

**Comment by pavan3008** [comment]
Thanks for pointing it out, Professor. I'll be re-requesting and reminding the reviewers, if there are any updates.
Please, review the new changes.

File: SmilesMiles/View/TaskListView.swift
Comment (wbraynen) [inline SmilesMiles/View/TaskListView.swift:33]: Nitpick, but easier to read if you format like this:
`})`

File: SmilesMiles/View/TaskListView.swift
Comment (wbraynen) [inline SmilesMiles/View/TaskListView.swift:37]: Why is the `Image` inside curlies here?  Can it just be the following:

```
Button...
Image...
Text...
```

Comment (wbraynen) [inline SmilesMiles/View/TaskListView.swift:37]: Elsewhere you have formatting like this:
```
Button(action: {
    isPresentingNewTrip = true
}) {
    Image(systemName: "plus.circle.fill")
```
I would format the same way here.  That way it's consistent and also easier to read.

File: SmilesMiles/View/TripView.swift
Comment (wbraynen) [inline SmilesMiles/View/TripView.swift:20]: Note that right now your code is not unit testable (unless you plan to unit-test your views).  We don't usually unit-test views because views, arguably, should be UI-tested (with automated UI tests, not automated unit tests).  Instead, we use _view models_.  You could put all the non-UI business-logic code in a view model and use the view model in your view and then unit-test the view model.

File: SmilesMilesTests/SmilesMilesTests.swift
Comment (wbraynen) [inline SmilesMilesTests/SmilesMilesTests.swift:34]: Why add this code if you don't have any unit tests.  It's a good idea to have unit tests though.  For that you would need to refactor your code for your views to use a view model.

File: SmilesMilesUITests/SmilesMilesUITests.swift
Comment (wbraynen) [inline SmilesMilesUITests/SmilesMilesUITests.swift:41]: Why add this?  Or else, you could actually add automated UI tests.

File: SmilesMilesUITests/SmilesMilesUITestsLaunchTests.swift
Comment (wbraynen) [inline SmilesMilesUITests/SmilesMilesUITestsLaunchTests.swift:31]: What does this test test?

**Comment by wbraynen** [comment]
@pavan3008 Once your PR has enough approvals for you to merge, then you’ll probably want to merge it using the “Squash and Merge” button.

I left a few more comments though.   Let’s try to get it merged today.  Sorry for the delay on my end.

Proposal 1: fix the one formatting issue I pointed out, remove your tests, and then add a follow-up PR to add unit tests.
Proposal 2: fix everything up in this PR.

How do you want to proceed?

Re. branch names: 
Your target branch: I would just call it “main” instead of “master” or “dev”.  Self-explanatory then because it is really your main branch.  It’s GitHub’s default anyway.
Your feature branch: Your feature branch isn’t named in a very informative way. Don’t worry about it in this PR, but your next working branch should be named better.  For example, `pavan/LAB-365-added-trips-ui`.

**Pull request closed without merging**
